pop-api/examples/fungibles/tests.rs

```diff
@@ -0,0 +1,831 @@
+use super::*;
+use drink::{
+	sandbox_api::assets_api::AssetsAPI,
+	session::{ContractBundle, Session},
+};
+use frame_support::assert_ok;
+use pop_api::{
+	primitives::{
+		ArithmeticError::Overflow,
+		Error::{Arithmetic, Module},
+		TokenId,
+	},
+	v0::fungibles::events::{Approval, Created, Transfer},
+};
+use pop_sandbox::{Balance, DevnetSandbox as Sandbox, ALICE, BOB};
+use scale::Encode;
+use std::sync::LazyLock;
+use utils::*;
+
+#[drink::contract_bundle_provider]
+enum BundleProvider {}
+
+const AMOUNT: Balance = MIN_BALANCE * 4;
+const MIN_BALANCE: Balance = 10_000;
+const TOKEN: TokenId = 1;
+
+static CONTRACT: LazyLock<ContractBundle> = LazyLock::new(|| BundleProvider::local().unwrap());
+
+/// Deployment and constructor method tests.
+
+#[drink::test(sandbox = Sandbox)]
+fn new_constructor_works(mut session: Session) {
+	let _ = env_logger::try_init();
+	// Fails to deploy contract with an used token ID.
+	let token = TOKEN + 1;
+	assert_ok!(session.sandbox().create(&token, &ALICE, MIN_BALANCE));
+	// `pallet-assets` returns `InUse` error.
+	assert_eq!(
+		deploy(
+			&mut session,
+			CONTRACT.clone(),
+			"new",
+			vec![token.to_string(), MIN_BALANCE.to_string()]
+		),
+		Err(into_psp22_custom(Module { index: 52, error: [5, 0] }))
+	);
+
+	// Deploy a new contract.
+	let contract = deploy(
+		&mut session,
+		CONTRACT.clone(),
+		"new",
+		vec![TOKEN.to_string(), MIN_BALANCE.to_string()],
+	)
+	.unwrap();
+	// Token exists after the deployment.
+	assert!(session.sandbox().asset_exists(&TOKEN));
+	// Successfully emit event.
+	assert_eq!(
+		last_contract_event(&session).unwrap(),
+		Created {
+			id: TOKEN,
+			creator: account_id_from_slice(contract.as_ref()),
+			admin: account_id_from_slice(contract.as_ref()),
+		}
+		.encode()
+		.as_slice()
+	);
+}
+
+#[drink::test(sandbox = Sandbox)]
+fn existing_constructor_works(mut session: Session) {
```

Comment:
Would prefer separate tests for each case to make tests shorter. There is no state setup so easy to separate and better demonstrates by test naming all the cases/conditions which are being checked.

Consider how simple and easy the contract reads. The tests should follow suit.

Reply:
https://github.com/r0gue-io/pop-node/pull/314